### PR TITLE
Adds support to indent with the tab key

### PR DESCRIPTION
**Pull request by blackjid**
Disables tab key navigation from textarea elements

### CODE REVIEW

**Comment by hkdobrev** [comment]
I guess you want that so you can easily type code in comment boxes, right?

Could you change it to use 4 spaces? It's more likely to be consistent when the comment is seen in various email clients and not taking so much space like 8 spaces.

**Comment by blackjid** [comment]
Exactly @hkdobrev, always struggled whit the tab key trying to indent when writing code snippets on comments...

File: extension/content.js [hunk 1]
```diff
@@ -294,6 +294,33 @@ function markMergeCommitsInList() {
 	});
 }
 
+function indentInput(el, size = 4) {
+	el.focus();
+	const text = " ";
+	const val = el.value;
+	if (typeof el.selectionStart === "number") {
+		const selStart = el.selectionStart;
+		const intendSize = size - el.selectionEnd % size || size;
```
Comment (hkdobrev) [inline extension/content.js]: Could you rename `indendSize` to `indentSize`?

File: extension/content.js [hunk 2]
```diff
@@ -294,6 +294,33 @@ function markMergeCommitsInList() {
 	});
 }
 
+function indentInput(el, size = 4) {
+	el.focus();
+	const text = " ";
+	const val = el.value;
+	if (typeof el.selectionStart === "number") {
+		const selStart = el.selectionStart;
```
Comment (hkdobrev) [inline extension/content.js]: Could you rename `selStart` to the more explicit `selectionStart`?

File: extension/content.js [hunk 3]
```diff
@@ -294,6 +294,33 @@ function markMergeCommitsInList() {
 	});
 }
 
+function indentInput(el, size = 4) {
+	el.focus();
+	const text = " ";
```
Comment (hkdobrev) [inline extension/content.js]: Could you use single quotes like everywhere else?

File: extension/content.js [hunk 4]
```diff
@@ -294,6 +294,33 @@ function markMergeCommitsInList() {
...
+	const val = el.value;
+	if (typeof el.selectionStart === "number") {
```
Comment (hkdobrev) [inline extension/content.js]: Single quotes here again.

File: readme.md [hunk 5]
```diff
@@ -26,6 +26,7 @@ My hope is that GitHub will notice and implement some of these much needed impro
 - Removes annoying hover effect in the repo file browser
 - Removes the comment box toolbar
 - Moves the dashboard organization switcher to the right column
+- Support indent with tab key in textareas
```
Comment (sindresorhus) [inline readme.md]: > Supports indenting with the tab key in textareas

Comment (blackjid) [inline readme.md]: do you want to me to squash everything in a single commit after that fix?

Comment (sindresorhus) [inline readme.md]: Yup

File: extension/content.js [hunk 6]
```diff
@@ -294,6 +294,33 @@ function markMergeCommitsInList() {
 	});
 }
 
+function indentInput(el, size = 4) {
+	el.focus();
+	const text = ' ';
```
Comment (sindresorhus) [inline extension/content.js]: No need for a variable for this. Just put it inline.

File: extension/content.js [hunk 7]
```diff
@@ -294,6 +294,33 @@ function markMergeCommitsInList() {
...
+	const val = el.value;
+	if (typeof el.selectionStart === 'number') {
```
Comment (sindresorhus) [inline extension/content.js]: Is `selectionStart` ever not a number?

File: extension/content.js [hunk 8]
```diff
@@ -294,6 +294,32 @@ function markMergeCommitsInList() {
 	});
 }
 
+function indentInput(el, size = 4) {
+	el.focus();
+	const val = el.value;
+	if (typeof el.selectionStart === 'number') {
```
Comment (sindresorhus) [inline extension/content.js]: Is `selectionStart` ever not a number?

File: extension/content.js [hunk 9]
```diff
@@ -294,6 +294,32 @@ function markMergeCommitsInList() {
...
+		const selectionStart = el.selectionStart;
+		const indentSize = size - el.selectionEnd % size || size;
```
Comment (sindresorhus) [inline extension/content.js]:
```js
const indentSize = (size - el.selectionEnd % size) || size;
```

For clarity.

File: extension/content.js (hunk 8 above)
Comment (blackjid) [inline extension/content.js]: no is not.. you are right... I refactor this code for a snippet from stackoverflow, ant I think it had support for IE11, that was a check for that...

File: extension/content.js [hunk 10]
```diff
@@ -294,6 +294,30 @@ function markMergeCommitsInList() {
 	});
 }
 
+function indentInput(el, size = 4) {
+	el.focus();
+	const val = el.value;
+	const selectionStart = el.selectionStart;
+	const indentSize = (size - el.selectionEnd % size) || size;
+	const indentationText = ' '.repeat(indentSize);
+	el.value = val.slice(0, selectionStart) + indentationText + val.slice(el.selectionEnd);
```
Comment (hkdobrev) [inline extension/content.js]: I'm sorry for seeing this later, but I think `value` is better than `val`. What do you think?

Comment (blackjid) [inline extension/content.js]: yes, you are right... 
:)

**Comment by sindresorhus** [comment]
@blackjid This is really awesome! I always wanted this and just ended up writing all my code blocks in my editor and copy-paste. Keep'em coming :)

**Pull request merged**